Documentation/ceph-nfs-crd.md

```diff
@@ -13,109 +14,96 @@ resource definition. This will spin up a cluster of
 via shared RADOS objects. The servers will be configured for NFSv4.1+ access only, as serving
 earlier protocols can inhibit responsiveness after a server restart.
 
-> **WARNING**: We do not recommend using NFS in Ceph v16.2.0 through v16.2.6. If you are using Ceph
-> v15, we encourage you to upgrade directly to Ceph Pacific v16.2.7.
-> [Upgrade steps are outlined below.](#upgrading-from-ceph-v15-to-v16)
+> **WARNING**: This document assumes you are using v16.2.7 or higher.
+> If you are using or wish to use an earlier than v16.2.7, see the documented
+> [known issue](#ceph-v1620-through-v1626) below.
 
 ## Samples
-The following sample assumes Ceph v16 and will create a two-node active-active cluster of NFS
-Ganesha gateways.
 
 ```yaml
 apiVersion: ceph.rook.io/v1
 kind: CephNFS
 metadata:
   name: my-nfs
-  namespace: rook-ceph
+  namespace: rook-ceph # namespace:cluster
 spec:
-  # For Ceph v15, the rados block is required. It is ignored for Ceph v16.
-  rados:
-    # RADOS pool where NFS configs are stored.
-    # In this example the data pool for the "myfs" filesystem is used.
-    # If using the object store example, the data pool would be "my-store.rgw.buckets.data".
-    # Note that this has nothing to do with where exported file systems or object stores live.
-    pool: myfs-data0
-    # RADOS namespace where NFS client recovery data is stored in the pool.
-    namespace: nfs-ns
-
   # Settings for the NFS server
   server:
-    # the number of active NFS servers
-    active: 2
-    # A key/value list of annotations
-    annotations:
-    #  key: value
-    # where to run the NFS server
+    active: 1
+
     placement:
-    #  nodeAffinity:
-    #    requiredDuringSchedulingIgnoredDuringExecution:
-    #      nodeSelectorTerms:
-    #      - matchExpressions:
-    #        - key: role
-    #          operator: In
-    #          values:
-    #          - mds-node
-    #  tolerations:
-    #  - key: mds-node
-    #    operator: Exists
-    #  podAffinity:
-    #  podAntiAffinity:
-    #  topologySpreadConstraints:
-
-    # The requests and limits set here allow the ganesha pod(s) to use half of one CPU core and 1 gigabyte of memory
+      nodeAffinity:
+        requiredDuringSchedulingIgnoredDuringExecution:
+          nodeSelectorTerms:
+          - matchExpressions:
+            - key: role
+              operator: In
+              values:
+              - nfs-node
+      topologySpreadConstraints:
+      tolerations:
+      - key: nfs-node
+        operator: Exists
+      podAffinity:
+      podAntiAffinity:
+
+    annotations:
```

Comment:
Some of these standard properties like annotations, labels, and resources seem better left for the example nfs.yaml, but they could be removed from this doc for simplicity.

Reply:
Since this documents the CRD, I think it is most helpful to have all possible configs listed here. I have found it confusing to understand where things go in CRDs when it is just the bullet point documentation. Even if they aren't doing anything in this example, I think readers can at least see more clearly that the option exists and where it goes hierarchically at a glance.